src/App.js

```diff
@@ -11,36 +12,187 @@ class App extends React.Component {
       cardDeck: makeShuffledDeck(),
       // currCards holds the cards from the current round
       currCards: [],
+      //index of the winner per each round
+      indexWinner: -1,
+      player1Score: 0,
+      player2Score: 0,
+      //the winner of the game in the end of 26 rounds
+      roundWinner: null,
+      gameStart: true,
+      //overall player score across multiple game sessions
+      overallPlayer1Score: 0,
+      overallPlayer2Score: 0,
     };
   }
 
   dealCards = () => {
-    // this.state.cardDeck.pop() modifies this.state.cardDeck array
+    // dealing cards
     const newCurrCards = [this.state.cardDeck.pop(), this.state.cardDeck.pop()];
+
+    this.setState(
+      {
+        currCards: newCurrCards,
+      },
+      () => {
+        this.determineWinner();
+      }
+    );
+  };
+
+  reset = () => {
+    //reset every state except overall player score across multiple game sessions
     this.setState({
-      currCards: newCurrCards,
+      cardDeck: makeShuffledDeck(),
+      currCards: [],
+      indexWinner: -1,
+      player1Score: 0,
+      player2Score: 0,
+      restart: false,
+      roundWinner: null,
     });
   };
 
-  render() {
-    // You can write JavaScript here, just don't try and set your state!
+  determineWinner = () => {
+    //compare rank of player 1 and player 2 card
+    //return winner, either index 0 (player 1) or 1 (player 2) or 99 (tie).
+    let temp = -1;
 
-    // You can access your current components state here, as indicated below
-    const currCardElems = this.state.currCards.map(({ name, suit }) => (
-      // Give each list element a unique key
-      <div key={`${name}${suit}`}>
-        {name} of {suit}
-      </div>
+    if (this.state.currCards === null) {
+      temp = -1;
+    } else if (this.state.currCards[0].rank > this.state.currCards[1].rank) {
+      temp = 0;
+    } else if (this.state.currCards[1].rank > this.state.currCards[0].rank) {
+      temp = 1;
+    } else {
+      temp = 99;
+    }
+
+    //trigger function to count the total score
+    this.setState(
+      {
+        indexWinner: temp,
+      },
+      () => {
+        this.countScore();
+      }
+    );
+  };
+
+  countScore = () => {
+    //counting score for each player
+    const tempCounter = [0, 0];
+    //if indexWinner is 0, add +1 to count index 0, vice versa
+    if (this.state.indexWinner === 0) {
+      tempCounter[0]++;
+    } else if (this.state.indexWinner === 1) {
+      tempCounter[1]++;
+    }
+    //update all the scores
+    this.setState((prevState) => ({
+      player1Score: prevState.player1Score + tempCounter[0],
+      player2Score: prevState.player2Score + tempCounter[1],
+      overallPlayer1Score: prevState.overallPlayer1Score + tempCounter[0],
+      overallPlayer2Score: prevState.overallPlayer2Score + tempCounter[1],
+    }));
```

Comment:
This whole section seems a bit overengineered to me :) 

You probably could do a simple: 

```suggestion
    else if (this.state.currCards[0].rank > this.state.currCards[1].rank) {
         this.setState((prevState) => ({
      player1Score: prevState.player1Score + 1,
      overallPlayer1Score: prevState.overallPlayer1Score + 1,
    }));
    } else if (this.state.currCards[1].rank > this.state.currCards[0].rank) {
        this.setState((prevState) => ({
      player2Score: prevState.player2Score + 1,
      overallPlayer2Score: prevState.overallPlayer2Score + 1,
    }));
    } else {
      <do some tie thing>
    }
```